Sources/SPMBuildCore/PluginInvocation.swift

```diff
@@ -440,3 +438,32 @@ struct PluginScriptRunnerOutput: Codable {
         let outputFilesDirectory: String
     }
 }
+
+extension ObservabilityMetadata {
+    public var fileLocation: FileLocation? {
+        get {
+            self[FileLocationKey.self]
+        }
+        set {
+            self[FileLocationKey.self] = newValue
+        }
+    }
+
+    private enum FileLocationKey: Key {
+        typealias Value = FileLocation
+    }
+}
+
+public struct FileLocation: Equatable, CustomStringConvertible {
+    let file: String
```

Comment:
Is this sometimes a name and sometimes a path?  Should it be normalized to an `AbsolutePath`?

Reply:
Looks good now.  Marking resolved.